### PR TITLE
[Icons]Core Icon Revision

### PR DESCRIPTION
**Pull request by Reqrefusion**
#13840

### CODE REVIEW

**Comment by maxwxyz** [comment]
IMO the wireframe and hidden line icon is still too similar.
Maybe the hidden line icon should indicate the triangulation?

**Comment by Reqrefusion** [comment]
> IMO the wireframe and hidden line icon is still too similar. Maybe the hidden line icon should indicate the triangulation?

It probably mixes it up more. I think changing the perfect made it much, much easier to understand. Also I don't understand this idea. Also, I looked at other programs and it usually happens like this in all of them.

**Comment by maxwxyz** [comment]
So you say this makes it easier to understand but you don't understand it? The hidden lines display the triangulated state as the geometry is rendered with Coin3D. IMO not very useful for working with solids but it's the great for meshes.

**Comment by Reqrefusion** [comment]
> So you say this makes it easier to understand but you don't understand it? The hidden lines display the triangulated state as the geometry is rendered with Coin3D. IMO not very useful for working with solids but it's the great for meshes.

Ok I won't lie. I never looked at this view because I thought it was similar to other CAD programs. Which I don't normally use. I've never used mesh or anything like that. Now I kneel before the greatness of FreeCad. Indeed, this appearance has turned into something useful that can be used.

**Comment by maxwxyz** [comment]
I like it now. Thanks.
Please remove src/Gui/Icons/edge-selection.svg this icon from the PR as this will be deleted, otherwise this will get conflicts.
Then please squash your commits.

**Comment by Reqrefusion** [comment]
Done, ready to merge.

File: src/Gui/Icons/button_add_all.svg
Comment (maxwxyz) [inline src/Gui/Icons/button_add_all.svg:1]: The lines and stroke widths are not aligned with the pixel grid, that's why it looks blurred 
image

File: src/Gui/Icons/accessories-calculator.svg
Comment (maxwxyz) [inline src/Gui/Icons/accessories-calculator.svg:1]: the elements in this icon are not aligned to the pixel grid and that is the reason why it appears blurry. You can view the grid under View -> Page grid.
All elements and stroke widths should be aligned.

File: src/Gui/Icons/camera-photo.svg
Comment (maxwxyz) [inline src/Gui/Icons/camera-photo.svg:1]: same with this icon: 
image

File: src/Gui/Icons/umf-measurement.svg
Comment (maxwxyz) [inline src/Gui/Icons/umf-measurement.svg:1]: also this one is not aligned but the original wasn't too. Shall I do a version of it?

File: src/Gui/Icons/debug-marker.svg
Comment (maxwxyz) [inline src/Gui/Icons/debug-marker.svg:1]: Also not aligned:

original, sharp: 
image

your version, blurry:
image

File: src/Gui/Icons/application-exit.svg
Comment (maxwxyz) [inline src/Gui/Icons/application-exit.svg:1]: There is a gap, it seems like your paths are angled:
image

Also the icon is not aligned to the pixel grid.

**Review by maxwxyz — CHANGES_REQUESTED** [review]
The resizing broke several alignments to the pixel grid so the smaller icons appear blurry. 
I've attached some screenshots to the before/after. As mentioned, the elements and strokes should be aligned to the pixel grid to prevent aliasing and scale better.
In Inkscape you can view the grid under View -> Page grid.

src/Gui/Icons/accessories-text-editor.svg this icon is a good example how it should be, as everything is aligned.

@kadet1090 @obelisk79 FYI

**Comment by kadet1090** [comment]
Yes, pixel alignment is crucial for the icons to render properly and be sharp so it must be fixed before merging.

File: src/Gui/Icons/accessories-calculator.svg
Comment (Reqrefusion) [reply src/Gui/Icons/accessories-calculator.svg:1]: Corrected.

File: src/Gui/Icons/camera-photo.svg
Comment (Reqrefusion) [reply src/Gui/Icons/camera-photo.svg:1]: Corrected.

File: src/Gui/Icons/umf-measurement.svg
Comment (Reqrefusion) [reply src/Gui/Icons/umf-measurement.svg:1]: An icon that contains too many small details will break whether we want it or not. I still made corrections for the outer edges. An icon that contains too many small details will break whether we want it or not. I still made corrections for the outer edges. Besides, it was already broken before, and this is an obvious fix. Even being perfectly centered is a big plus. Icon alignment comes before pixel alignment.

File: src/Gui/Icons/button_add_all.svg
Comment (Reqrefusion) [reply src/Gui/Icons/button_add_all.svg:1]: Corrected

File: src/Gui/Icons/debug-marker.svg
Comment (Reqrefusion) [reply src/Gui/Icons/debug-marker.svg:1]: Corrected

File: src/Gui/Icons/application-exit.svg
Comment (Reqrefusion) [reply src/Gui/Icons/application-exit.svg:1]: Corrected

File: src/Gui/Icons/umf-measurement.svg
Comment (maxwxyz) [reply src/Gui/Icons/umf-measurement.svg:1]: perfectly centered is not always preferred, some things must be aligned optically.
Best demo is a triangle / play button: when it's aligned perfectly centered it looks too far to the left.

Comment (Reqrefusion) [reply src/Gui/Icons/umf-measurement.svg:1]: It is absolutely necessary for the margins to be the same, this is what I mean. The distance from left to right is the same there too. I don't extend it.

File: src/Gui/Icons/utilities-terminal.svg
[the remark this answers is not in the document]
Comment (Reqrefusion) [reply src/Gui/Icons/utilities-terminal.svg:1]: Yes, I looked at all mentioned.

**Comment by Reqrefusion** [comment]
Done, ready to merge.

**Comment by chennes** [comment]
@maxwxyz could you please approve (or not, as the case may be!) the change you requested?

**Comment by maxwxyz** [comment]
@kadet1090 @obelisk79 could you check as well and comment?

**Comment by chennes** [comment]
@obelisk79 if you would like to comment please do so, otherwise we will merge next week.

**Review by obelisk79 — CHANGES_REQUESTED** [review]

**Comment by Reqrefusion** [comment]
dagViewVisible
I don't think this eye looks disproportionate. But I still revert the change.

**Comment by maxwxyz** [comment]
The stepped line effect on the view icons is still there. Could you use the old perspective for them?

**Comment by Reqrefusion** [comment]
> The stepped line effect on the view icons is still there. Could you use the old perspective for them?

I don't understand what you mean by the stepping line effect.

**Comment by maxwxyz** [comment]
grafik
This is on a HiDPI screen so a normal screen would look worse. IMO just keep the old ones and properly adjust sizing if necessary.

**Comment by maxwxyz** [comment]
Plus the perspective is off on almost every line, so maybe it's best to keep the old ones 
grafik

**Comment by Reqrefusion** [comment]
DrawStyleAsIs
DrawStyleFlatLines
DrawStyleHiddenLine
DrawStyleNoShading
DrawStylePoints
DrawStyleShaded
DrawStyleWireFrame
I would definitely try at another time, but I'm leaving the group to return later. A change of perspective is definitely necessary. Step line effect is easy to fix, but I don't have time for the other one yet.

**Comment by Reqrefusion** [comment]
If there is no objection, it means there is no problem. I have dealt with all objections. Please merge if you don't see a problem.

**Comment by chennes** [comment]
@maxwxyz can you give a final look?

**Comment by maxwxyz** [comment]
@chennes you can merge this

**Pull request merged**